stackrox/stackrox
# Swap collections UI autocomplete dropdowns with plain text inputs

## Description

Removes the collection autocomplete dropdowns and replaces them with plain text inputs, since the autocomplete feature will not ship as part of 3.74 and the dropdowns are confusing without it.

I outright deleted the SelectAutocomplete component in this PR, since it is dead code now. I'll link this PR in the ticket for the feature though so that the path to recovery is clear.

## Checklist

## Testing Performed

Verification of unit and e2e tests.

Manual testing:

From the collection form, select a "by name" rule from the dropdown. The input fields should default to "exact match" and a plain text input should appear next to it. Entering a value into the text field should return the correct results on the right side panel.
![image](https://user-images.githubusercontent.com/1292638/213525823-5d34ad6d-7e31-4701-a3ed-65aee097c9a0.png)
![image](https://user-images.githubusercontent.com/1292638/213525862-65ee9c3c-5279-49b7-ad48-d84d7f4a199b.png)

From the collection form, select a "by label" rule from the dropdown. There should be a single plain text input field below. Entering a valid label should automatically return the correct results on the right side panel.
![image](https://user-images.githubusercontent.com/1292638/213528727-d530a381-688f-401d-bb57-57fc7c2e874c.png)

![image](https://user-images.githubusercontent.com/1292638/213526434-3b88970c-53ec-4685-bf26-8a96a5debfce.png)
